### PR TITLE
Using CLI11 ini files for configuration

### LINKED ISSUE #175: Use CLI11 ini file rather than json config file
This would simplify the code to a single definition of the options and default values. Any configuration message input could be accepted as a separate json file.

### PR DESCRIPTION
**Pull request by rerpha**
### Description of work
(Breaking change)

Uses `.ini` files when starting the forwarder instead of raw JSON. this means the actual streams to be added initially can be cut down. Also CLI11 has more error checking and ways to set defaults. 

Test coverage will decrease slightly as the json parsing stuff has been removed. There is no easy way to test CLI11s functionality nor should we as it's an external library but unfortunately this means that some lines of code have been added and can't be covered. 

### Issue

Closes #175 

### Acceptance Criteria

Check that all tests pass (including system tests)

### Unit Tests

Removed some unit tests which were responsible for checking if default values were set as this should be handled by CLI11 now. 

### Other

Updated documentation as new configuration files will now need to be made. 

---

## Code Review (To be filled in by the reviewer only)

- [ ] Is the code of an acceptable quality?
- [ ] Do the changes function as described and is it robust?

---

## Nominate for Group Code Review (Anyone can nominate it)
Indicate if you think the code should be reviewed in a Thursday code review session.

- [ ] Recommend for group code review

Also, nominate it on the code_review Slack channel (does someone want to automate this?).

### CODE REVIEW

File: src/MainOpt.cpp [hunk 1]
```diff
@@ -127,6 +113,16 @@ std::pair<int, std::unique_ptr<MainOpt>> parse_opt(int argc, char **argv) {
...
+  App.add_option("--conversion-threads", opt.MainSettings.ConversionThreads,
+                 "Conversion threads", true)
+      App.add_option("--conversion-worker-queue-size",
+                     opt.MainSettings.ConversionWorkerQueueSize,
+                     "Conversion worker queue size", true)
+          App.add_option("--main-poll-interval",
+                         opt.MainSettings.MainPollInterval,
+                         "Main Poll interval", true)
+              App.set_config("-c,--config-file", "",
+                             "Read configuration from an ini file", false);
```
Comment (mattclarke) [inline src/MainOpt.cpp]: Weird indentation

Comment (mattclarke) [reply src/MainOpt.cpp]: At least on GitHub

File: system-tests/config-files/forwarder_config_fake_epics.ini [hunk 2]
```diff
@@ -0,0 +1,3 @@
+broker = //localhost:9092
+config-topic = //localhost:9092/TEST_forwarderConfig
+streams-json = ./forwarder_config_fake_epics.json
```
Comment (matthew-d-jones) [inline system-tests/config-files/forwarder_config_fake_epics.ini]: Missing new line

File: system-tests/config-files/forwarder_config_idle_updates.ini [hunk 3]
```diff
@@ -0,0 +1,3 @@
+broker = //localhost:9092
+config-topic = //localhost:9092/TEST_forwarderConfig
+streams-json = ./forwarder_config_idle_updates.json
```
Comment (matthew-d-jones) [inline system-tests/config-files/forwarder_config_idle_updates.ini]: Missing new line

File: system-tests/config-files/forwarder_config.ini [hunk 4]
```diff
@@ -0,0 +1,3 @@
+broker = //localhost:9092
+config-topic = //localhost:9092/TEST_forwarderConfig
+streams-json = ./forwarder_config.json
```
Comment (matthew-d-jones) [inline system-tests/config-files/forwarder_config.ini]: Missing new line

File: docker_launch.sh [hunk 5]
```diff
@@ -6,4 +6,4 @@ source /forwarder/activate_run.sh
 
 echo "Launching forwarder"
 
-/forwarder/forward-epics-to-kafka --config-file=${CONFIG_FILE:="forwarder_config.json"} --fake-pv-period=${FAKE_PV_PERIOD_MS:="0"} --pv-update-period=${IDLE_PV_UPDATE_PERIOD:="0"}
+/forwarder/forward-epics-to-kafka --config-file=${CONFIG_FILE:="forwarder_config.ini"} --fake-pv-period=${FAKE_PV_PERIOD_MS:="0"} --pv-update-period=${IDLE_PV_UPDATE_PERIOD:="0"} -v7
```
Comment (matthew-d-jones) [inline docker_launch.sh]: Could you set the log level in the ini rather than hard code here please?

File: src/MainOpt.cpp [hunk 6]
```diff
@@ -140,15 +180,17 @@ std::pair<int, std::unique_ptr<MainOpt>> parse_opt(int argc, char **argv) {
...
     try {
-      opt.parse_json_file(opt.ConfigurationFile);
+      opt.MainSettings.StreamsInfo = opt.parseStreamsJson(opt.StreamsFile);
     } catch (std::exception const &e) {
       LOG(Sev::Warning, "Can not parse configuration file: {}", e.what());
       ret.first = 1;
       return ret;
     }
   }
+  LOG(Sev::Info, "-------config topic = {}-------",
```
Comment (rerpha) [inline src/MainOpt.cpp]: remove this

**Pull request merged**